managed-ledger/src/main/java/org/apache/bookkeeper/mledger/Entry.java

```diff
@@ -48,7 +50,18 @@
     Position getPosition();
 
     /**
-     * Release the resources allocated for this entry
+     * Increments ref-count of entry data
+     */
+    void retain();
+    
+    /**
+     * Recycles entry instance 
+     */
+    void recycle();
```

Comment:
Since there is already a ref-counting semantic on the class, can we take advantage of it and automatically recycle when ref-count goes to 0?